internal/rbd/rbd_util_test.go

```diff
@@ -283,3 +283,43 @@ func TestStrategicActionOnLogFile(t *testing.T) {
 		})
 	}
 }
+
+func TestIsKrbdFeatureSupported(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name        string
+		featureName string
+		isSupported bool
+		err         error
+	}{
+		{
+			name:        "supported feature",
+			featureName: "layering",
+			isSupported: true,
+			err:         nil,
+		},
+		{
+			name:        "not supported feature",
+			featureName: "journaling",
+			isSupported: false,
+			err:         nil,
+		},
+		{
+			name:        "unrecognized feature",
+			featureName: "xyz",
+			isSupported: false,
+			err:         librbd.ErrNotFound,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			supported, err := isKrbdFeatureSupported(tt.featureName)
```

Comment:
This now is dependent on the system where the tests are running. If the rbd kernel module is not (and can not be) loaded, this will always fail. You have to think of an other way to test the functions, without depending on the rbd kernel module.

Reply:
I agree. Should we discard the unit test?

Reply:
Or is there a way to `modprobe rbd` prior to running unit tests from the CI scripts?

Reply:
Always have unit tests for helper functions. Maybe initialize the supported features once with a global variable, and set the variable to something pre-defined in the tests.

unit tests should be able to run as non-root while developing ceph-csi.

Reply:
Apologies, I didn't follow completely,
It is `getKrbdSupportedFeatures` job to get supported features (load rbd module, read the /sys file ...) now this unit test is for `isKrbdFeatureSupported`. How does defining the global variable for supported features is going to help? I mean how do I bypass the call to `getKrbdSupportedFeatures` ?

Reply:
yes, the tests can not rely on `getKrbdSupportedFeatures` (which only needs to be called once anyway). So, I'd suggest checking a global variable in `isKrbdFeatureSupported` (which could be returned by `getKrbdSupportedFeatures` if set earlier). Unit tests can then set the global variable, making `getKrbdSupportedFeatures` do (almost) nothing, and the test will work everywhere.

Reply:
Sure, Thanks applied now.